torch/_dynamo/utils.py

```diff
@@ -1371,3 +1371,11 @@ def tensor_shape_should_be_static(
     if not is_tensor:
         return True, TensorStaticReason.NOT_TENSOR
     return False, None
+
+
+def normalize_attr_name(name):
+    # e.g. replace abc.xyz[123].qkv with abc.xyz_123.qkv
+    name = re.sub(r"\[(\d+)\]", r"_\g<1>", name)
+    # e.g. replace abc.xyz_123.qkv with abc_xyz_123_qkv
+    name = re.sub(r"[^a-zA-Z0-9]", "_", name)
+    return name.lower()
```

Comment:
Why lower?

Reply:
Later on, we use this to normalize lookups, and there were cases where placeholder names and attr names were case mismatched.

Reply:
I'm skeptical. In the context here, there's no requirement that names have to be consistent against calls, because we still uniqueify it below. You seem to imply that you actually need similar things to match.